openedx/features/edly/tests/test_models.py

```diff
@@ -0,0 +1,113 @@
+        organizations = EdlyOrganization.objects.filter(name=org_name)
+
+        # Make sure an entry (and only one entry) is saved for Organization
+        assert len(organizations) == 1
+
+    def test_site_configuration_post_update_receiver(self):
+        """
+        Test that entry is updated in Organization model.
+        """
+        org_name = 'updated test org name'
+
+        # add SiteConfiguration to database
+        organization = EdlyOrganizationFactory()
+
+        organization.name = org_name
+        organization.save()
+
+        # Verify an entry to Organization was updated.
+        organizations = EdlyOrganization.objects.filter(name=org_name)
+
+        assert len(organizations) == 1
+        # Verify an entry to Organization was updated is filtered
+        assert organizations[0].name == org_name
+
+    def test_slug_alphanumerice_validation(self):
+        org_slug = 'test $%3org slug'
+        organization = EdlyOrganizationFactory(slug=org_slug)
+        expect_exception = 'Only alphanumeric characters are allowed.'
+
+        with pytest.raises(Exception) as excinfo:
+            organization.full_clean()
+
+        assert expect_exception in str(excinfo.value)
+
+
+class SubOrganizationTests(TestCase):
+    """
+    Tests for SubOrganization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(SubOrganizationTests, cls).setUpClass()
+        cls.lms_site = SiteFactory()
+        cls.studio_site = SiteFactory()
+
+    def test_sub_organization_post_save_receiver(self):
+        """
+        Test that entry is added to SubOrganization model.
+        """
+        # add SubOrganization to database
+        sub_organization = EdlySubOrganizationFactory.create(lms_site=self.lms_site)
+
+        # Verify an entry to SubOrganization was added.
+        sub_organization_data = EdlySubOrganization.objects.filter(lms_site=sub_organization.lms_site)
+
+        # Make sure an entry (and only one entry) is saved for SubOrganization
+        assert len(sub_organization_data) == 1
+
+    def test_sub_organization_post_update_receiver(self):
+        """
+        Test that entry is updated in SubOrganization model.
+        """
+
+        org_name = 'updated test org name'
+
+        # add SiteConfiguration to database
+        edly_organization = EdlyOrganizationFactory()
+
+        edly_organization.name = org_name
+        edly_organization.save()
+
+        # add SubOrganization to database
+        sub_organization = EdlySubOrganizationFactory.create(studio_site=self.studio_site)
+
+        sub_organization.edly_organization = edly_organization
+        sub_organization.save()
+
+        # Verify an entry to SubOrganization was updated.
+        sub_organization_data = EdlySubOrganization.objects.filter(studio_site=sub_organization.studio_site)
+
+        assert len(sub_organization_data) == 1
+        # Verify an entry to SubOrganization was updated is filtered
```

Comment:
Update docstring the scantance is not correct. Please update all docscrting, comments and remove any unnecessary comments.

Reply:
Done